File: backend/app/api/v1/community.py
import csv
import io
from datetime import date

from flask import Blueprint, Response, request
from flask_jwt_extended import current_user, verify_jwt_in_request
from flask_restful import Api, Resource

from app.auth.decorators import permission_required
from app.extensions import db
from app.models.audit import AuditLog
from app.models.community import CommunityPage, Member, MemberNote
from app.models.taxonomy import Topic
from app.schemas.community import (
    CommunityPageInputSchema,
    CommunityPageSchema,
    CommunityPageStatusInputSchema,
    MemberAdminUpdateSchema,
    MemberJoinInputSchema,
    MemberNoteInputSchema,
    MemberSchema,
    MemberStatusInputSchema,
)
from app.schemas.media import MediaSchema
from app.services.audit import log_action
from app.services.content_blocks import sanitize_content_blocks
from app.services.newsletter import upsert_subscriber
from app.utils.filtering import apply_country_or_region_filter, apply_equality_filters, apply_search
from app.utils.pagination import paginate
from app.utils.responses import ApiError, success_response

community_bp = Blueprint("community", __name__)
api = Api(community_bp)

member_schema = MemberSchema()
page_schema = CommunityPageSchema()

_EXPORT_COLUMNS = [
    "First Name", "Last Name", "Email", "Country", "Organization", "Professional Title",
    "Status", "Membership Type", "Interests", "Joined Date", "Source", "Directory Opt-In",
]

# A repeat public Join with one of these existing statuses reactivates the
# record rather than creating a duplicate.
_REACTIVATABLE_STATUSES = ("left", "inactive")
# These already represent "a request/membership is on file" — acknowledge,
# never duplicate, never silently reactivate.
_ALREADY_PRESENT_STATUSES = ("active", "paused", "pending")
# Deliberate admin actions. A public Join must never silently reverse
# these, and — to avoid leaking internal moderation state to an anonymous
# requester — the response never distinguishes this case from a fresh
# signup's generic acknowledgement.
_ADMIN_SUPPRESSED_STATUSES = ("declined", "archived")

_JOIN_WRITABLE_FIELDS = (
    "first_name", "last_name", "professional_title", "organization_name", "short_bio",
    "website_url", "linkedin_url", "country_code", "city", "referral_note", "source", "acquisition",
)
_ADMIN_WRITABLE_FIELDS = (
    "first_name", "last_name", "email", "professional_title", "organization_name", "short_bio",
    "website_url", "linkedin_url", "city", "country_code", "membership_type", "referral_note", "source",
    "community_updates_opt_in", "directory_opt_in", "admin_tags", "person_id", "profile_image_media_id",
)


def _require_active_user():
    verify_jwt_in_request()
    if not current_user or not current_user.is_active:
        raise ApiError("Account is inactive or no longer exists.", 403, code="forbidden")
    return current_user


def _require_export():
    user = _require_active_user()
    if not user.has_permission("community.export"):
        raise ApiError("You do not have permission to export members.", 403, code="forbidden")
    return user


def _get_member_or_404(member_id):
    member = db.session.get(Member, member_id)
    if member is None:
        raise ApiError("Member not found.", 404, code="not_found")
    return member


def _resolve_topics(slugs):
    if not slugs:
        return []
    return Topic.query.filter(Topic.slug.in_(slugs)).all()


def _get_community_page():
    page = db.session.get(CommunityPage, 1)
    if page is None:
        raise ApiError("Community page not initialized.", 500, code="not_found")
    return page


def _apply_join_fields(member, data):
    for field in _JOIN_WRITABLE_FIELDS:
        if field in data:
            setattr(member, field, data[field])
    member.consent_given = True
    member.consent_at = db.func.now()
    member.newsletter_opt_in = bool(data.get("subscribe_newsletter"))


def _subscribe_to_newsletter(member, data):
    upsert_subscriber(
        email=member.email,
        first_name=member.first_name,
        last_name=member.last_name,
        country_code=member.country_code,
        placement="community-join",
        acquisition=data.get("acquisition"),
    )
    db.session.commit()


def _build_member_query():
    query = Member.query.order_by(Member.created_at.desc())
    query = apply_equality_filters(query, Member, request.args, ["status", "membership_type"])
    query = apply_search(query, Member, request.args, ["first_name", "last_name", "email", "organization_name"])
    query = apply_country_or_region_filter(query, Member, request.args)

    directory_opt_in = request.args.get("directory_opt_in")
    if directory_opt_in is not None:
        query = query.filter(Member.directory_opt_in == (directory_opt_in == "true"))

    interest = request.args.get("interest")
    if interest:
        query = query.join(Member.interests).filter(Topic.slug == interest)

    return query


class MemberJoinResource(Resource):
    def post(self):
        data = MemberJoinInputSchema().load(request.get_json(silent=True) or {})
        email = data["email"].strip().lower()
        interest_slugs = data.pop("interest_slugs", [])

        existing = Member.query.filter_by(email=email).first()

        if existing is not None:
            if existing.status in _ADMIN_SUPPRESSED_STATUSES:
                return success_response(
                    {"message": "Thank you for your interest in Women Shaping Futures. Our team will be in touch if there's anything further needed."}
                )
            if existing.status in _ALREADY_PRESENT_STATUSES:
                return success_response(
                    {"message": "It looks like you're already part of the WSF community. If you'd like to update your details, please contact us."}
                )
            # _REACTIVATABLE_STATUSES: welcome them back.
            member = existing
            previous_status = member.status
            member.status = "active"
            member.activated_at = db.func.now()
            member.left_at = None
            _apply_join_fields(member, data)
            member.interests = _resolve_topics(interest_slugs)
            db.session.commit()
            if member.newsletter_opt_in:
                _subscribe_to_newsletter(member, data)
            return success_response({"message": "Welcome back! Your membership has been reactivated."})

        member = Member(email=email, status="active", activated_at=db.func.now())
        _apply_join_fields(member, data)
        member.interests = _resolve_topics(interest_slugs)
        db.session.add(member)
        db.session.commit()
        if member.newsletter_opt_in:
            _subscribe_to_newsletter(member, data)
        return success_response({"message": "Welcome to the Women Shaping Futures community!"}, status=201)


def build_public_member_payload(member):
    return {
        "id": member.id,
        "name": member.full_name,
        "professionalTitle": member.professional_title,
        "organizationName": member.organization_name,
        "shortBio": member.short_bio,
        "country": {"code": member.country.code, "name": member.country.name} if member.country else None,
        "interests": [t.name for t in member.interests],
        "profileImage": MediaSchema().dump(member.profile_image) if member.profile_image else None,
    }


class PublicMemberDirectoryResource(Resource):
    def get(self):
        query = Member.query.filter_by(status="active", directory_opt_in=True).order_by(Member.first_name)
        query = apply_country_or_region_filter(query, Member, request.args)
        interest = request.args.get("interest")
        if interest:
            query = query.join(Member.interests).filter(Topic.slug == interest)
        result = paginate(query)
        return success_response([build_public_member_payload(m) for m in result["items"]], meta=result["meta"])


def build_public_community_page(page):
    return {
        "hero": {
            "heading": page.hero_heading,
            "description": page.hero_description,
            "media": MediaSchema().dump(page.hero_media) if page.hero_media else None,
        },
        "introContent": page.intro_content or [],
        "benefits": page.benefits or [],
        "whoForText": page.who_for_text,
        "howToJoinText": page.how_to_join_text,
        "cta": {"heading": page.cta_heading, "description": page.cta_description, "buttonLabel": page.cta_button_label},
        "faq": page.faq or [],
        "seo": page.seo or {},
    }


class PublicCommunityResource(Resource):
    """One composite call so the public /community page never has to fan
    out into separate content + metrics requests.
    """

    def get(self):
        page = db.session.get(CommunityPage, 1)
        if page is None or page.status != "published":
            raise ApiError("Community page not available.", 404, code="not_found")

        member_count = Member.query.filter_by(status="active").count()
        metrics = None
        if member_count > 0:
            country_count = (
                db.session.query(Member.country_code)
                .filter(Member.status == "active", Member.country_code.isnot(None))
                .distinct()
                .count()
            )
            metrics = {"memberCount": member_count, "countryCount": country_count}

        return success_response({"page": build_public_community_page(page), "metrics": metrics})


class MemberListResource(Resource):
    @permission_required("community.manage")
    def get(self):
        result = paginate(_build_member_query(), member_schema)
        return success_response(result["items"], meta=result["meta"])


class MemberDetailResource(Resource):
    @permission_required("community.manage")
    def get(self, member_id):
        return success_response(member_schema.dump(_get_member_or_404(member_id)))

    @permission_required("community.manage")
    def patch(self, member_id):
        member = _get_member_or_404(member_id)
        data = MemberAdminUpdateSchema().load(request.get_json(silent=True) or {})

        if "email" in data:
            normalized = data["email"].strip().lower()
            conflict = Member.query.filter(Member.email == normalized, Member.id != member.id).first()
            if conflict:
                raise ApiError("Another member already uses this email address.", 409, code="conflict")
            data["email"] = normalized

        interest_slugs = data.pop("interest_slugs", None)
        directory_changed = "directory_opt_in" in data and data["directory_opt_in"] != member.directory_opt_in

        for field in _ADMIN_WRITABLE_FIELDS:
            if field in data:
                setattr(member, field, data[field])
        if interest_slugs is not None:
            member.interests = _resolve_topics(interest_slugs)

        db.session.commit()
        log_action(current_user, "member.update", "Member", member.id)
        if directory_changed:
            log_action(
                current_user, "member.directory_visibility_change", "Member", member.id,
                changes={"directoryOptIn": member.directory_opt_in},
            )
        return success_response(member_schema.dump(member))

    @permission_required("community.manage")
    def delete(self, member_id):
        member = _get_member_or_404(member_id)
        if member.status not in ("pending", "declined"):
            raise ApiError(
                "Only Pending or Declined members can be deleted. Archive established members instead.",
                409, code="delete_restricted",
            )
        if member.notes:
            raise ApiError(
                "This member has internal notes and can't be deleted. Archive it instead.",
                409, code="delete_restricted",
            )
        db.session.delete(member)
        db.session.commit()
        log_action(current_user, "member.delete", "Member", member_id)
        return success_response(None, status=204)


class MemberStatusResource(Resource):
    @permission_required("community.manage")
    def patch(self, member_id):
        member = _get_member_or_404(member_id)
        data = MemberStatusInputSchema().load(request.get_json(silent=True) or {})
        previous_status = member.status
        member.status = data["status"]

        if previous_status != "active" and member.status == "active":
            member.activated_at = db.func.now()
            member.left_at = None
        if member.status in ("left", "inactive", "archived") and previous_status not in ("left", "inactive", "archived"):
            member.left_at = db.func.now()

        db.session.commit()
        log_action(
            current_user, "member.status_change", "Member", member.id,
            changes={"from": previous_status, "to": member.status},
        )
        return success_response(member_schema.dump(member))


class MemberNoteListResource(Resource):
    @permission_required("community.manage")
    def post(self, member_id):
        member = _get_member_or_404(member_id)
        data = MemberNoteInputSchema().load(request.get_json(silent=True) or {})
        note = MemberNote(member=member, user=current_user, body=data["body"])
        db.session.add(note)
        db.session.commit()
        log_action(current_user, "member.note_add", "Member", member.id)
        return success_response(member_schema.dump(member), status=201)


class MemberHistoryResource(Resource):
    @permission_required("community.manage")
    def get(self, member_id):
        _get_member_or_404(member_id)
        entries = (
            AuditLog.query.filter_by(entity_type="Member", entity_id=str(member_id))
            .order_by(AuditLog.created_at.desc())
            .all()
        )
        return success_response(
            [
                {
                    "id": e.id,
                    "action": e.action,
                    "changes": e.changes,
                    "user": e.user.full_name if e.user else None,
                    "createdAt": e.created_at.isoformat() if e.created_at else None,
                }
                for e in entries
            ]
        )


class MemberExportResource(Resource):
    def get(self):
        _require_export()

        query = _build_member_query()
        members = query.limit(20000).all()

        buffer = io.StringIO()
        writer = csv.writer(buffer)
        writer.writerow(_EXPORT_COLUMNS)
        for m in members:
            writer.writerow(
                [
                    m.first_name,
                    m.last_name,
                    m.email,
                    m.country.name if m.country else "",
                    m.organization_name or "",
                    m.professional_title or "",
                    m.status,
                    m.membership_type,
                    ", ".join(t.name for t in m.interests),
                    m.applied_at.isoformat() if m.applied_at else "",
                    m.source or "",
                    "Yes" if m.directory_opt_in else "No",
                ]
            )

        log_action(current_user, "member.export", "Member", None, changes={"count": len(members)})
        response = Response(buffer.getvalue(), mimetype="text/csv")
        response.headers["Content-Disposition"] = f"attachment; filename=wsf-members-{date.today().isoformat()}.csv"
        return response


class CommunityPageResource(Resource):
    @permission_required("community.manage")
    def get(self):
        return success_response(page_schema.dump(_get_community_page()))

    @permission_required("community.manage")
    def patch(self):
        page = _get_community_page()
        data = CommunityPageInputSchema().load(request.get_json(silent=True) or {})
        if "intro_content" in data:
            data["intro_content"] = sanitize_content_blocks(data["intro_content"])
        for field, value in data.items():
            setattr(page, field, value)
        db.session.commit()
        log_action(current_user, "community_page.update", "CommunityPage", page.id)
        return success_response(page_schema.dump(page))


class CommunityPageStatusResource(Resource):
    @permission_required("community.manage")
    def patch(self):
        page = _get_community_page()
        data = CommunityPageStatusInputSchema().load(request.get_json(silent=True) or {})
        previous_status = page.status
        page.status = data["status"]
        db.session.commit()
        log_action(
            current_user, "community_page.status_change", "CommunityPage", page.id,
            changes={"from": previous_status, "to": page.status},
        )
        return success_response(page_schema.dump(page))


api.add_resource(MemberJoinResource, "/join")
api.add_resource(PublicMemberDirectoryResource, "/directory")
api.add_resource(PublicCommunityResource, "/public")
api.add_resource(MemberListResource, "/members")
api.add_resource(MemberExportResource, "/members/export")
api.add_resource(MemberDetailResource, "/members/<int:member_id>")
api.add_resource(MemberStatusResource, "/members/<int:member_id>/status")
api.add_resource(MemberNoteListResource, "/members/<int:member_id>/notes")
api.add_resource(MemberHistoryResource, "/members/<int:member_id>/history")
api.add_resource(CommunityPageResource, "/page")
api.add_resource(CommunityPageStatusResource, "/page/status")

File: frontend/src/api/community.js
import { apiClient, USE_MOCK } from './client'
import { delay } from './mockUtils'
import { mapMediaRef } from '../utils/media'

// No dedicated Community mock dataset exists yet (this module is new) — a
// small self-contained fixture keeps mock mode functional without
// growing the shared mock/admin.js file for a shape nothing else reads.
const MOCK_PAGE = {
  id: 1,
  hero_heading: 'The WSF Community',
  hero_description:
    'A global home for ambitious, career-driven women — connect with members across industries, career stages, and countries.',
  hero_media: null,
  intro_content: [],
  benefits: [
    { title: 'Opportunities', description: 'Surface jobs, opportunities, and events curated for our community.' },
    { title: 'Insights', description: 'Leadership and business insights from across the WSF platform.' },
    { title: 'Connection', description: 'A global network of women shaping their industries and futures.' },
  ],
  who_for_text: 'Women at any career stage — early career to executive leadership — across every industry and country.',
  how_to_join_text: "Fill out the form below. It's free, takes two minutes, and membership is open to all.",
  cta_heading: 'Join the community',
  cta_description: 'Tell us a bit about yourself and we will be in touch.',
  cta_button_label: 'Join WSF',
  faq: [],
  status: 'published',
  seo: {},
  created_at: '2026-01-01T00:00:00Z',
  updated_at: '2026-01-01T00:00:00Z',
}

const MOCK_MEMBERS = []

export function mapCommunityPage(p) {
  if (!p) return null
  return {
    id: p.id,
    heroHeading: p.hero_heading || '',
    heroDescription: p.hero_description || '',
    heroMedia: mapMediaRef(p.hero_media),
    heroMediaId: p.hero_media_id || null,
    introContent: Array.isArray(p.intro_content) ? p.intro_content : [],
    benefits: Array.isArray(p.benefits) ? p.benefits : [],
    whoForText: p.who_for_text || '',
    howToJoinText: p.how_to_join_text || '',
    ctaHeading: p.cta_heading || '',
    ctaDescription: p.cta_description || '',
    ctaButtonLabel: p.cta_button_label || '',
    faq: Array.isArray(p.faq) ? p.faq : [],
    status: p.status || 'draft',
    seo: p.seo || {},
    createdAt: p.created_at || null,
    updatedAt: p.updated_at || null,
  }
}

export function mapMember(m) {
  if (!m) return null
  return {
    id: m.id,
    firstName: m.first_name || '',
    lastName: m.last_name || '',
    fullName: m.full_name || `${m.first_name || ''} ${m.last_name || ''}`.trim(),
    email: m.email || '',
    professionalTitle: m.professional_title || '',
    organizationName: m.organization_name || '',
    shortBio: m.short_bio || '',
    websiteUrl: m.website_url || '',
    linkedinUrl: m.linkedin_url || '',
    profileImage: mapMediaRef(m.profile_image),
    profileImageMediaId: m.profile_image_media_id || null,
    country: m.country ? { code: m.country.code, name: m.country.name, region: m.country.region } : null,
    countryCode: m.country?.code || '',
    city: m.city || '',
    status: m.status || 'active',
    membershipType: m.membership_type || 'Community Member',
    source: m.source || '',
    referralNote: m.referral_note || '',
    appliedAt: m.applied_at || null,
    activatedAt: m.activated_at || null,
    leftAt: m.left_at || null,
    interests: Array.isArray(m.interests) ? m.interests.map((t) => ({ id: t.id, slug: t.slug, name: t.name })) : [],
    interestSlugs: Array.isArray(m.interests) ? m.interests.map((t) => t.slug) : [],
    consentGiven: !!m.consent_given,
    consentAt: m.consent_at || null,
    newsletterOptIn: !!m.newsletter_opt_in,
    communityUpdatesOptIn: !!m.community_updates_opt_in,
    directoryOptIn: !!m.directory_opt_in,
    person: m.person ? { id: m.person.id, slug: m.person.slug, name: m.person.name, title: m.person.title } : null,
    personId: m.person_id || null,
    adminTags: Array.isArray(m.admin_tags) ? m.admin_tags : [],
    notes: Array.isArray(m.notes)
      ? m.notes.map((n) => ({ id: n.id, body: n.body, user: n.user?.full_name || null, createdAt: n.created_at }))
      : [],
    createdAt: m.created_at || null,
    updatedAt: m.updated_at || null,
  }
}

function mapPublicMember(m) {
  return {
    id: m.id,
    name: m.name,
    professionalTitle: m.professionalTitle || '',
    organizationName: m.organizationName || '',
    shortBio: m.shortBio || '',
    country: m.country || null,
    interests: m.interests || [],
    profileImage: mapMediaRef(m.profileImage),
  }
}

// ---------------------------------------------------------------------------
// Public
// ---------------------------------------------------------------------------

export async function joinCommunity(payload) {
  const body = {
    firstName: payload.firstName,
    lastName: payload.lastName,
    email: payload.email,
    countryCode: payload.countryCode,
    interestSlugs: payload.interestSlugs || [],
    consentGiven: !!payload.consentGiven,
    professionalTitle: payload.professionalTitle || undefined,
    organizationName: payload.organizationName || undefined,
    shortBio: payload.shortBio || undefined,
    websiteUrl: payload.websiteUrl || undefined,
    linkedinUrl: payload.linkedinUrl || undefined,
    city: payload.city || undefined,
    referralNote: payload.referralNote || undefined,
    source: payload.source || undefined,
    subscribeNewsletter: !!payload.subscribeNewsletter,
    acquisition: payload.acquisition || undefined,
  }
  if (!USE_MOCK) {
    try {
      const { data } = await apiClient.post('/community/join', body)
      return { success: true, message: data.message }
    } catch (err) {
      return { success: false, message: err.apiError?.message || 'Something went wrong. Please try again.' }
    }
  }
  MOCK_MEMBERS.push({ ...body, status: 'active' })
  return delay({ success: true, message: 'Welcome to the Women Shaping Futures community!' }, 500)
}

export async function fetchPublicCommunity() {
  if (!USE_MOCK) {
    const { data } = await apiClient.get('/community/public')
    return {
      page: {
        heroHeading: data.page.hero.heading || '',
        heroDescription: data.page.hero.description || '',
        heroMedia: mapMediaRef(data.page.hero.media),
        introContent: data.page.introContent || [],
        benefits: data.page.benefits || [],
        whoForText: data.page.whoForText || '',
        howToJoinText: data.page.howToJoinText || '',
        ctaHeading: data.page.cta.heading || '',
        ctaDescription: data.page.cta.description || '',
        ctaButtonLabel: data.page.cta.buttonLabel || '',
        faq: data.page.faq || [],
        seo: data.page.seo || {},
      },
      metrics: data.metrics || null,
    }
  }
  return delay({
    page: {
      heroHeading: MOCK_PAGE.hero_heading, heroDescription: MOCK_PAGE.hero_description, heroMedia: null,
      introContent: [], benefits: MOCK_PAGE.benefits, whoForText: MOCK_PAGE.who_for_text,
      howToJoinText: MOCK_PAGE.how_to_join_text, ctaHeading: MOCK_PAGE.cta_heading,
      ctaDescription: MOCK_PAGE.cta_description, ctaButtonLabel: MOCK_PAGE.cta_button_label, faq: [], seo: {},
    },
    metrics: null,
  })
}

export async function fetchCommunityDirectory(params = {}) {
  if (!USE_MOCK) {
    const { data } = await apiClient.get('/community/directory', { params })
    return (data || []).map(mapPublicMember)
  }
  return delay([])
}

// ---------------------------------------------------------------------------
// Admin — members
// ---------------------------------------------------------------------------

export async function fetchMembers(params = {}) {
  if (!USE_MOCK) {
    const { data } = await apiClient.get('/community/members', { params })
    return { ...data, items: data.items.map(mapMember) }
  }
  return delay({ items: [], meta: { page: 1, per_page: 20, total: 0, total_pages: 0, has_next: false, has_prev: false } })
}

export async function fetchMember(id) {
  if (!USE_MOCK) {
    const { data } = await apiClient.get(`/community/members/${id}`)
    return mapMember(data)
  }
  return delay(null)
}

function buildMemberPayload(payload) {
  return {
    firstName: payload.firstName,
    lastName: payload.lastName,
    email: payload.email,
    professionalTitle: payload.professionalTitle || undefined,
    organizationName: payload.organizationName || undefined,
    shortBio: payload.shortBio || undefined,
    websiteUrl: payload.websiteUrl || undefined,
    linkedinUrl: payload.linkedinUrl || undefined,
    city: payload.city || undefined,
    countryCode: payload.countryCode || undefined,
    membershipType: payload.membershipType || undefined,
    interestSlugs: payload.interestSlugs,
    referralNote: payload.referralNote || undefined,
    source: payload.source || undefined,
    communityUpdatesOptIn: payload.communityUpdatesOptIn,
    directoryOptIn: payload.directoryOptIn,
    adminTags: payload.adminTags,
    personId: payload.personId === '' || payload.personId == null ? undefined : Number(payload.personId),
    profileImageMediaId: payload.profileImage?.id || undefined,
  }
}

export async function updateMember(id, payload) {
  const { data } = await apiClient.patch(`/community/members/${id}`, buildMemberPayload(payload))
  return mapMember(data)
}

export async function updateMemberStatus(id, status) {
  const { data } = await apiClient.patch(`/community/members/${id}/status`, { status })
  return mapMember(data)
}

export async function addMemberNote(id, body) {
  const { data } = await apiClient.post(`/community/members/${id}/notes`, { body })
  return mapMember(data)
}

export async function deleteMember(id) {
  await apiClient.delete(`/community/members/${id}`)
}

export async function fetchMemberHistory(id) {
  if (!USE_MOCK) {
    const { data } = await apiClient.get(`/community/members/${id}/history`)
    return data
  }
  return delay([])
}

export async function exportMembers(params = {}) {
  const response = await apiClient.get('/community/members/export', { params, responseType: 'blob' })
  const url = window.URL.createObjectURL(new Blob([response.data], { type: 'text/csv' }))
  const link = document.createElement('a')
  link.href = url
  link.setAttribute('download', `wsf-members-${new Date().toISOString().slice(0, 10)}.csv`)
  document.body.appendChild(link)
  link.click()
  link.remove()
  window.URL.revokeObjectURL(url)
}

// ---------------------------------------------------------------------------
// Admin — community page
// ---------------------------------------------------------------------------

export async function fetchCommunityPage() {
  if (!USE_MOCK) {
    const { data } = await apiClient.get('/community/page')
    return mapCommunityPage(data)
  }
  return delay(mapCommunityPage(MOCK_PAGE))
}

export async function updateCommunityPage(payload) {
  const body = {
    heroHeading: payload.heroHeading || undefined,
    heroDescription: payload.heroDescription || undefined,
    heroMediaId: payload.heroMedia?.id || undefined,
    introContent: payload.introContent,
    benefits: payload.benefits,
    whoForText: payload.whoForText || undefined,
    howToJoinText: payload.howToJoinText || undefined,
    ctaHeading: payload.ctaHeading || undefined,
    ctaDescription: payload.ctaDescription || undefined,
    ctaButtonLabel: payload.ctaButtonLabel || undefined,
    faq: payload.faq,
    seo: payload.seo,
  }
  if (!USE_MOCK) {
    const { data } = await apiClient.patch('/community/page', body)
    return mapCommunityPage(data)
  }
  Object.assign(MOCK_PAGE, { hero_heading: body.heroHeading ?? MOCK_PAGE.hero_heading })
  return delay(mapCommunityPage(MOCK_PAGE))
}

export async function updateCommunityPageStatus(status) {
  if (!USE_MOCK) {
    const { data } = await apiClient.patch('/community/page/status', { status })
    return mapCommunityPage(data)
  }
  MOCK_PAGE.status = status
  return delay(mapCommunityPage(MOCK_PAGE))
}
